design/0005-circuit-breaking.md

```diff
@@ -0,0 +1,150 @@
+# Title
+
+- Contribution Name: circuit-breaking
+- Implementation Owner: @Ashish-Bansal, Linkerd Team
+- Start Date: 2020-06-14
+- Target Date: N.A.
+- Linkerd Issue:
+  [linkerd/linkerd2#2846](https://github.com/linkerd/linkerd2/issues/2846)
+- Reviewers: @adleong
+
+## Summary
+
+[summary]: #summary
+
+We would like to implement circuit breaking functionality which would allow end
+users to configure mesh such that they can limit the impact of failures from
+faulty service over others and prevent cascading issues. By configuring circuit
+breaking, they will be able to remove failing service instances from load
+balancing and survive through it in a more graceful manner.
+
+## Problem Statement
```

Comment:
It is super helpful in the problem statement to be very explicit about the concrete use case you're imagining. Would you mind providing a diagram with all the services, their behavior and walk through step by step for what should happen? This'll really help prioritize what features we'd like to see in the design section.